pkg/ecdsa/dkg/protocol_test.go

```diff
@@ -0,0 +1,279 @@
+package dkg
+
+import (
+	"fmt"
+	"github.com/keep-network/keep-core/pkg/crypto/ephemeral"
+	"github.com/keep-network/keep-core/pkg/internal/testutils"
+	"github.com/keep-network/keep-core/pkg/protocol/group"
+	"reflect"
+	"testing"
+)
+
+// TODO: This file contains unit tests that stress each protocol phase
+//       separately. We should also develop integration tests just like
+//       we did for Random Beacon's DKG protocol. This will require
+//       to refactor the `pkg/internal/dkgtest` package in the way that
+//       it supports the ECDSA DKG as well.
+
+func TestGenerateEphemeralKeyPair(t *testing.T) {
+	groupSize := 3
+	dishonestThreshold := 0
+
+	members := initializeEphemeralKeyPairGeneratingMembersGroup(
+		dishonestThreshold,
+		groupSize,
+	)
+
+	// Generate ephemeral key pairs for each group member.
+	messages := make(map[group.MemberIndex]*ephemeralPublicKeyMessage)
+	for _, member := range members {
+		message, err := member.generateEphemeralKeyPair()
+		if err != nil {
+			t.Fatal(err)
+		}
+		messages[member.id] = message
+	}
+
+	// Assert that each member has a correct state.
+	for _, member := range members {
+		// Assert the right key pairs count is stored in the member's state.
+		expectedKeyPairsCount := groupSize - 1
+		actualKeyPairsCount := len(member.ephemeralKeyPairs)
+		if expectedKeyPairsCount != actualKeyPairsCount {
+			t.Fatalf(
+				"[member:%v] wrong number of stored ephemeral key pairs\n"+
+					"expected: [%v]\n"+
+					"actual:   [%v]",
+				member.id,
+				expectedKeyPairsCount,
+				actualKeyPairsCount,
+			)
+		}
```

Comment:
What about adding`AssertIntegersEqual` to `testutils.go` to reduce the boilerplate a bit?

Reply:
See e62c8610